File: web/src/types/notice.ts
export interface NoticeItem {
  id: number
  title: string
  content: string
  enabled: boolean
  sort_order: number
  starts_at: string | null
  ends_at: string | null
  created_at: string
  updated_at: string
}

export interface NoticePayload {
  title: string
  content: string
  enabled: boolean
  sort_order: number
  starts_at: string | null
  ends_at: string | null
}

File: server/internal/model/notice.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Notice struct {
	ID        uint64         `gorm:"primaryKey" json:"id"`
	Title     string         `gorm:"size:120;not null" json:"title"`
	Content   string         `gorm:"type:text" json:"content"`
	Enabled   bool           `gorm:"not null;default:false;index" json:"enabled"`
	SortOrder int            `gorm:"not null;default:0;index" json:"sort_order"`
	StartsAt  *time.Time     `gorm:"index" json:"starts_at"`
	EndsAt    *time.Time     `gorm:"index" json:"ends_at"`
	CreatedAt time.Time      `gorm:"index" json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}
